### DISCUSSION TITLE
useLocaleHead doesn't change or its not reactive from 8.0.1 version.

### DISCUSSION THREAD

**Discussion by mini-op** [nuxt-modules/i18n#2765] [Q&A]
![image](https://github.com/nuxt-modules/i18n/assets/77070946/cb04397f-cc81-492a-b834-e0bbd8f9e4fa)
I have this in my app.vue file

after upgrading to 8.0.1 or anything else above from 8.0.0 version, the lang and dir attribute is not getting updated when i set locale from useI18n() unless I change page. 

I read the docs and change logs but not sure if am I missing something or not

**Comment by mini-op** [comment]
Here is the reproduction:
version 8.1.0
https://stackblitz.com/edit/bobbiegoede-nuxt-i18n-starter-cpbimm?file=components%2FLangSwitcher.vue

version 8.0.1
https://stackblitz.com/edit/bobbiegoede-nuxt-i18n-starter-td21q6?file=components%2FLangSwitcher.vue

version 8.0.0
https://stackblitz.com/edit/bobbiegoede-nuxt-i18n-starter-9rzhrz?file=components%2FLangSwitcher.vue

**Comment by pmnzt** [comment]
Can confirm this, yesterday fixed this issue of useLocaleHead not being reactive, in one of my friend's projects. He was using "@nuxtjs/i18n": "^8.1.0"


and when i downgraded to "@nuxtjs/i18n": "^8.0.0".
the issue got fixed.

That's all i can provide for now, i shouldn't share his project here without his permission, i might provide minimal reproduction if somehow i got time for it.

but thankfully maintainers might come across this and check the OP's reproduction.

**Comment by mini-op** [comment]
Based on this: https://github.com/nuxt-modules/i18n/issues/2771
In order to set locale, we have to import and use `setLocale()` instead of `locale`
